packages/profile-sync-controller/src/sdk/user-storage.ts

```diff
@@ -200,6 +201,47 @@ export class UserStorage {
     }
   }
 
+  async #batchUpsertUserStorageWithAlreadyHashedAndEncryptedEntries(
```

Comment:
Future - I don't like how we've shoved all the user storage logic in a class, this bloats the SDK class and makes it harder to test things individually.

Similar to the authentication SDK, we should split out `services` from the user storage SDK.

Reply:
@Prithpal-Sooriya Agreed! Added that to my to-do list!